lib/pwwka/receiver.rb

```diff
@@ -29,13 +29,22 @@ def self.subscribe(handler_klass, queue_name, routing_key: "#.#", block: true)
             receiver.ack(delivery_info.delivery_tag)
             logf "Processed Message on %{queue_name} -> %{payload}, %{routing_key}", queue_name: queue_name, payload: payload, routing_key: delivery_info.routing_key
           rescue => e
+            error_options = {
+               queue_name: queue_name,
+                  payload: payload,
+              routing_key: delivery_info.routing_key,
+                exception: e
+            }
             if Pwwka.configuration.requeue_on_error && !delivery_info.redelivered
-              logf "Retrying an Error Processing Message on %{queue_name} -> %{payload}, %{routing_key}: %{exception}: %{backtrace}", queue_name: queue_name, payload: payload, routing_key: delivery_info.routing_key, exception: e, backtrace: e.backtrace.join(';'), at: :error
+              log_error "Retrying an Error Processing Message", error_options
               receiver.nack_requeue(delivery_info.delivery_tag)
             else
-              logf "Error Processing Message on %{queue_name} -> %{payload}, %{routing_key}: %{exception}: %{backtrace}", queue_name: queue_name, payload: payload, routing_key: delivery_info.routing_key, exception: e, backtrace: e.backtrace.join(';'), at: :error
+              log_error "Error Processing Message", error_options
               receiver.nack(delivery_info.delivery_tag)
             end
+            unless Pwwka.configuration.keep_alive_on_handler_klass_exceptions?
+              raise Interrupt,"Exiting due to exception #{e.inspect}"
```

Comment:
This is the change.  FWIW, without #43 this would've been incredibly difficult to test in any real way.

The reason we throw `Interrupt` is to a) rescue it below and shutdown gracefully, but b) throwing a `StandardError` triggers Bunny's exception handling callback.  We can replace that callback with our own, but it would have to throw an `Exception` to actually escape this block.  So, it seemed more expedient to do it here.